io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java

```diff
@@ -578,7 +597,31 @@ public Instance next() {
 			}
 
 			Instance instance = iterator.next();
-			return new StreamGmlInstance(instance, totalFeaturesProcessed++);
+
```

Comment:
@stempler Could you help me understand how is it possible that for a total or
`totalFeaturesProcessed: 754` and `additionalFeatureProcessed: 1844` processed here with a sum of 2598, 
HS shows the following numbers:
![image](https://github.com/halestudio/hale/assets/67077116/794bc430-af71-4cab-96f4-904fa7302491)
Thank you

Reply:
I would suggest to add a unit test where a WFS FeatureCollection file with additional objects is read. That doesn't allow to test the pagination, but if you include at least one object with a duplicate GML ID you can use this to test:

1. The correct detection of the "additional objects"
2. If your mechanism for detecting GML ID duplicates works as expected

Once it has been ruled out that a problem is hidden there the paging behavior can be analysed.